### PR TITLE
Auth proxy status fix

### PR DESCRIPTION
**Pull request by jooseppi-luna**
# Description
Currently, the authorization proxy server status is always failed in csm-operator. This is for a couple reasons:
1) We are attempting to calculate the status of the daemonset, but since auth proxy has no daemonset, it always fails. To fix this, the daemonset is no longer considered when checking the auth proxy deployment.
2) Beyond that, the handleDeploymentUpdate function is now looking for a csm label on the deployment objects. If it doesn't find it, the handler exits without doing anything else. To fix this, labels were added to all of the auth proxy deployment objects.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1143 |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration
Ran auth proxy e2e test. Will run full e2e tests before fixed-release-v1.4.2 is released.

### CODE REVIEW

**Review by JacobGros — APPROVED** [review]
LGTM

**Review by bharathsreekanth — APPROVED** [review]

File: pkg/utils/status.go [hunk 1]
```diff
@@ -389,25 +390,38 @@ func calculateState(ctx context.Context, instance *csmv1.ContainerStorageModule,
...
+	nodeStatusGood := true
+	newStatus.State = constants.Succeeded
 	// TODO: Currently commented this block of code as the API used to get the latest deployment status is not working as expected
 	// TODO: Can be uncommented once this issues gets sorted out
 	controllerReplicas, controllerStatus, controllerErr := getDeploymentStatus(ctx, instance, r)
 	if controllerErr != nil {
 		log.Infof("error from getDeploymentStatus: %s", controllerErr.Error())
 	}
 
+	// Auth proxy has no daemonset. Putting this if/else in here and setting nodeStatusGood to true by
```
[the remark this answers is not in the document]
Comment (jooseppi-luna) [reply pkg/utils/status.go:402]: @bharathsreekanth thanks for bringing that up, once v1.4.2 is out I will scope out the refactoring work and work on getting that into backlog

[the remark this answers is not in the document]
Comment (jooseppi-luna) [reply pkg/utils/status.go:402]: The TODO block above is from 6+ months ago but hopefully we will be able to address that as well

File: pkg/utils/status.go [hunk 2]
```diff
@@ -945,3 +961,104 @@ func observabilityStatusCheck(ctx context.Context, instance *csmv1.ContainerStor
...
+	}
+
+	checkFn := func(deployment *appsv1.Deployment) bool {
+		return deployment.Status.ReadyReplicas == *deployment.Spec.Replicas
+	}
+
+	for _, deployment := range deploymentList.Items {
+		deployment := deployment
+		switch deployment.Name {
+		case "authorization-ingress-nginx-controller":
```
Comment (shaynafinocchiaro) [inline pkg/utils/status.go:1000]: Will this check pass if the namespace isn't authorization?

Comment (JacobGros) [reply pkg/utils/status.go:1000]: thanks for catching that.
That will be a known issue for 1.9.2 and [fixed ](https://github.com/dell/csm-operator/tree/auth-namespace-fix)in 1.10

**Pull request merged**